### PR TITLE
MBMPO Cartpole

### PR DESCRIPTION
**Pull request by michaelzhiluo**
Adds MBMPO Cartpole (Discrete Space Support) as an example.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/mbmpo/tests/test_mbmpo.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def test_mbmpo_compilation(self):
         for _ in framework_iterator(config, frameworks="torch"):
             trainer = mbmpo.MBMPOTrainer(
                 config=config,
-                env="ray.rllib.examples.env.mbmpo_env.PendulumWrapper")
+                env="ray.rllib.examples.env.mbmpo_env.CartPoleWrapper")
```
Comment (ericl) [inline rllib/agents/mbmpo/tests/test_mbmpo.py:30]: We should still test both right?

**Review by ericl — CHANGES_REQUESTED** [review]

File: rllib/tuned_examples/mbmpo/cartpole-mbmpo.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    config:
+        # Only supported in torch right now.
+        framework: torch
+        #horizon: 200
+        num_envs_per_worker: 20
+        inner_adaptation_steps: 1
+        maml_optimizer_steps: 8
+        gamma: 0.99
+        lambda: 1.0
+        lr: 0.001
```
Comment (ericl) [inline rllib/tuned_examples/mbmpo/cartpole-mbmpo.yaml:16]: Add link in algorithms.rst?

File: rllib/agents/mbmpo/tests/test_mbmpo.py (hunk 1 above)
Comment (michaelzhiluo) [reply rllib/agents/mbmpo/tests/test_mbmpo.py:30]: Turns out MBMPO is not tested anymore (in MAML higher PR) b/c torch MAML uses higher.

File: rllib/tuned_examples/mbmpo/cartpole-mbmpo.yaml (hunk 2 above)
Comment (michaelzhiluo) [reply rllib/tuned_examples/mbmpo/cartpole-mbmpo.yaml:16]: Sgtm

**Comment by ericl** [comment]
Ping @michaelzhiluo

File: rllib/tuned_examples/mbmpo/cartpole-mbmpo.yaml (hunk 2 above)
Comment (michaelzhiluo) [reply rllib/tuned_examples/mbmpo/cartpole-mbmpo.yaml:16]: @ericl Done

**Comment by michaelzhiluo** [comment]
@ericl Added docs

**Review by sven1977 — APPROVED** [review]
Looks good. Just the replaced Pendulum test should be reinstated.
Can we not add higher to the testing requirements and run both envs in test_mbmpo?

**Review by ericl — APPROVED** [review]

**Comment by ericl** [comment]
+1 to add higher to test req if possible...

**Pull request merged**